src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py

```diff
@@ -0,0 +1,225 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+
+from msrestazure.azure_exceptions import CloudError
+from azure.cli.core._util import CLIError
+from azure.cli.core.commands.arm import is_valid_resource_id
+from ._client_factory import (get_devtestlabs_management_client)
+from azure.mgmt.devtestlabs.models.gallery_image_reference import GalleryImageReference
+import azure.cli.core.azlogging as azlogging
+
+logger = azlogging.get_az_logger(__name__)
+devtestlabs_management_client = get_devtestlabs_management_client(None)
+
+
+def get_complex_argument_processor(expanded_arguments, assigned_arg, model_type):
+    """
+    Return a validator which will aggregate multiple arguments to one complex argument.
+    """
+    def _expension_valiator_impl(namespace):
+        """
+        The validator create a argument of a given type from a specific set of arguments from CLI
+        command.
+        :param namespace: The argparse namespace represents the CLI arguments.
+        :return: The argument of specific type.
+        """
+        ns = vars(namespace)
+        kwargs = dict((k, ns[k]) for k in ns if k in set(expanded_arguments))
+
+        setattr(namespace, assigned_arg, model_type(**kwargs))
+
+    return _expension_valiator_impl
+
+
+def validate_location(namespace):
```

Comment:
I'd love that but `lab` resource is inside the `resource group`. That said, to query the information of the lab i need to provide resource group that makes it non defaultable. 

Reference: https://azure.github.io/projects/apis/?url=https://azure.github.io/specs/arm-devtestlabs/2016-05-15/swagger/DTL.json#!/Lab/Lab_GetResource